corehq/apps/reports/commtrack/ledgers_by_location.py

```diff
@@ -0,0 +1,121 @@
+from collections import namedtuple
+from dimagi.utils.decorators.memoized import memoized
+from django.utils.translation import ugettext as _, ugettext_lazy
+from django.utils.decorators import method_decorator
+from no_exceptions.exceptions import Http400
+
+from corehq.toggles import SUPPLY_REPORTS
+from corehq.apps.commtrack.models import StockState
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.products.models import SQLProduct
+from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn
+
+from .const import STOCK_SECTION_TYPE
+from corehq.apps.reports.generic import GenericTabularReport
+from corehq.apps.reports.commtrack.standard import CommtrackReportMixin
+
+LocationLedger = namedtuple('Row', "location stock")
+
+
+class LedgersByLocationDataSource(object):
+    """
+    Data source for a report showing ledger values at each location.
+
+                   | Product 1 | Product 2 |
+        Location 1 |        76 |        11 |
+        Location 2 |       132 |        49 |
+    """
+
+    def __init__(self, domain, section_id, page_start, page_size):
+        self.domain = domain
+        self.section_id = section_id
+        self.page_start = page_start
+        self.page_size = page_size
+
+    @property
+    @memoized
+    def products(self):
+        if SQLProduct.objects.filter(domain=self.domain).count() > 20:
+            raise Http400("This domain has too many products.")
+        return list(SQLProduct.objects.filter(domain=self.domain).order_by('name'))
+
+    def get_locations_queryset(self):
+        return (SQLLocation.objects
```

Comment:
I don't know too much about supply, but do you need to filter out archived locations here?